libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -689,13 +690,31 @@ void Compass::_detect_backends(void)
 #endif
 
 #if HAL_WITH_UAVCAN
-    if ((AP_BoardConfig::get_can_enable() != 0) && (hal.can_mgr != nullptr))
-    {
-        if((_backend_count < COMPASS_MAX_BACKEND) && (_compass_count < COMPASS_MAX_INSTANCES))
-        {
-            printf("Creating AP_Compass_UAVCAN\n\r");
-            _backends[_backend_count] = new AP_Compass_UAVCAN(*this);
-            _backend_count++;
+    if (AP_BoardConfig_CAN::get_can_enable() != 0) {
+        hal.scheduler->delay(2000);
```

Comment:
All external compasses get higher priority than internal ones. UAVCAN always had that too and should continue to.
By the way, you can turn off any detected compass, just there is a limit on detecting just 3 - and that is why we try to detect external ones first.